core/connectivity/ConnectivityService.h

```diff
@@ -48,9 +48,7 @@ class ConnectivityService
 
     virtual bool isConnected() = 0;
 
-    virtual bool publish(std::shared_ptr<Message> outboundMessage, bool persistent = false) = 0;
-
-    virtual void setUncontrolledDisonnectMessage(std::shared_ptr<Message> outboundMessage, bool persistent = false) = 0;
+    virtual bool publish(std::shared_ptr<MqttMessage> outboundMessage, bool persistent) = 0;
```

Comment:
You should leave this defaulted to false or remove the argument completely since I don't think it is ever used with value true

Reply:
I will remove it. Pure virtual methods in C++ shouldn't have default parameter values.